lib/pacemaker/pcmk_sched_constraints.c

```diff
@@ -332,62 +332,57 @@ unpack_simple_rsc_order(xmlNode * xml_obj, pe_working_set_t * data_set)
         return FALSE;
     }
 
-    invert_bool = order_is_symmetrical(xml_obj, kind, NULL);
-
-    id_then = crm_element_value(xml_obj, XML_ORDER_ATTR_THEN);
     id_first = crm_element_value(xml_obj, XML_ORDER_ATTR_FIRST);
-
-    action_then = crm_element_value(xml_obj, XML_ORDER_ATTR_THEN_ACTION);
-    action_first = crm_element_value(xml_obj, XML_ORDER_ATTR_FIRST_ACTION);
-
-    instance_then = crm_element_value(xml_obj, XML_ORDER_ATTR_THEN_INSTANCE);
-    instance_first = crm_element_value(xml_obj, XML_ORDER_ATTR_FIRST_INSTANCE);
-
-    if (action_first == NULL) {
-        action_first = RSC_START;
-    }
-    if (action_then == NULL) {
-        action_then = action_first;
-    }
-
```

Comment:
With all the config checking moved to the front, it might make sense to have it all then functionized.  That function could return true/false, unpack_simple_rsc_order could be turned into a void function, and then the caller of unpack_simple_rsc_order only call it if the config checking function returned true.  Like this:
```
if (any_sets == FALSE) {
    if (new_checking_function_here(xml_obj, data_set) == FALSE) {
        return FALSE;
    }

    unpack_simple_rsc_order(xml_obj, data_set);
    return TRUE;
}
```

Reply:
After reading through the code, this may be easier said than done.